### ISSUE TITLE
Issue with set Due date cannot be displayed

### ISSUE THREAD

**Issue by pskokovic** [berti92/mega_calendar#93]
**Environment:**

```
Redmine version     4.1.1.stable
Ruby version        2.5.1-p57 (2018-03-29) [x86_64-linux-gnu]
Rails version       5.2.4.2              
Redmine plugins:
  mega_calendar     1.7.2
```
**Description:** 
1. Create new issue (make sure **Due date is empty**)
2. Edit just created issue, set some Due date (e.g. 2 days in the future)
3. On Submit the following is displayed to user:
```
    Successful update.
    404
    The page you were trying to access doesn't exist or has been removed.
```
From now on, it is not possible to display this issue, but it can be edited.

production.log file contains the following:
```
Started GET "/issues/7?issue_count=6&issue_position=1&next_issue_id=6" for 1.2.3.4 at 2020-05-20 08:49:18 +0000.
Processing by IssuesController#show as HTML.
  Parameters: {"issue_count"=>"6", "issue_position"=>"1", "next_issue_id"=>"6", "id"=>"7"}
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3]   Rendered mailer/_issue.text.erb (6.7ms)
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3]   Rendered mailer/issue_edit.text.erb within layouts/mailer (7.8ms)
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3]   Rendering mailer/issue_edit.html.erb within layouts/mailer
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3]   Rendered mailer/_issue.html.erb (0.7ms)
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3]   Rendered mailer/issue_edit.html.erb within layouts/mailer (1.5ms)
[ActiveJob] [ActionMailer::DeliveryJob] [974b5901-74db-494f-a0b4-e9e6da7640e3] Performed ActionMailer::DeliveryJob (Job ID: 974b5901-74db-494f-a0b4-e9e6da7640e3) from Async(mailers) in 26.0ms
  Current user: user (id=6)
  Rendering plugins/mega_calendar/app/views/issues/show.html.erb within layouts/base
  Rendered issues/_action_menu.html.erb (2.1ms)
  Rendered issue_relations/_form.html.erb (0.7ms)
  Rendered issues/_relations.html.erb (1.1ms)
  Rendered plugins/mega_calendar/app/views/issues/show.html.erb within layouts/base (13.5ms)
Missing template, responding with 404: Missing partial issues/_history, application/_history with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :rsb]}. Searched in:
  * "/opt/redmine/plugins/mega_calendar/app/views"
  * "/opt/redmine/app/views"

  Rendering common/error.html.erb within layouts/base
  Rendered common/error.html.erb within layouts/base (0.1ms)
Completed 404 Not Found in 64ms (Views: 7.4ms | ActiveRecord: 24.2ms).
```

**Note:** This does not happen if Due date is set while creating new issue. It occurs only after editing, even if later Due date is cleared.

**Comment by kitsune242** [comment]
I have the same problem like you :(

**Comment by Interpunktor** [comment]
The same here...

**Environment:**
```
  Redmine version                4.1.1.stable.19939
  Ruby version                   2.5.1-p57 (2018-03-29) [x86_64-linux-gnu]
  Rails version                  5.2.4.2
  Environment                    production
Redmine plugins:
  mega_calendar                  1.7.2
  redmine_issue_dynamic_edit     0.7.2
  redmine_wysiwyg_editor         0.16.2
```

**Comment by berti92** [comment]
I have a look at it this weekend

**Comment by goto110** [comment]
Hi everyone.
I'm newby in redmine, but i think i found way to fix this problem.
```
cp mega_calendar/app/views/issues/_form.html.erb mega_calendar/app/views/issues/_history.html.erb # Not sure

In file mega_calendar/app/views/issues/show.html.erb replace "4.0" => "4." throughout the text
In file mega_calendar/app/views/issues/_attributes.html.erb replace "4.0" => "4." throughout the text

In file mega_calendar/app/views/issues/show.html.erb replace code:
<% if @journals.present? %>
<div id="history">
<h3><%=l(:label_history)%></h3>
<%= render :partial => 'history', :locals => { :issue => @issue, :journals => @journals } %>
</div>
<% end %>
With this:
 <div id="history">
<%= render_tabs issue_history_tabs, issue_history_default_tab %>
</div>

<!--
<% if @journals.present? %>
<div id="history">
<h3><%=l(:label_history)%></h3>
<%= render :partial => 'history', :locals => { :issue => @issue, :journals => @journals } %>
</div>
<% end %>
-->
```

**Comment by berti92** [comment]
Should be fixed in the current version.
